v2/contribute/decisions/oauth/0015-storing-client-secret.mdx

```diff
@@ -0,0 +1,49 @@
+---
+id: "0015"
+title: Only store the hashed client secret
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Only store the hashed client secret
+
+<DecisionHeader status="proposed" lastUpdate="2023-05-11" created="2023-05-11" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+Some other providers let you see the client secret multiple times.
+
+## Considered Options
+
+* Only store the hash and not support viewing it later
+* Store the secrets as plaintext
+* Store the secrets encrypted
+
```

Comment:
Another option is to store the hash and the first few chars of the secret. So users can kind of know which secret belongs to which client if they have many pairs. But i don;t think we need to do this. But worth mentioning